### DISCUSSION TITLE
Toleration Format to add under Process for k8's executor

### DISCUSSION THREAD

**Discussion by Booabi** [nextflow-io/nextflow#4269] [Q&A] [answered]
Nextflow version: 23.04.2
Nextflow.config File:

```
process {
  pod =  [[label: 'component', value: 'nextflow-job'], [toleration: 'example-key=namespace:NoSchedule']]
  executor = 'k8s'
}

k8s {
  namespace = 'nextflow'
  serviceAccount = 'nextflow-sa'
}
```

```
Error:
N E X T F L O W  ~  version 23.04.2
Launching `https://github.com/nextflow-io/rnaseq-nf` [stupefied_poitras] DSL2 - revision: 88b8ef803a [master]
 R N A S E Q - N F   P I P E L I N E
 ===================================
 transcriptome: /home/boopathi/.nextflow/assets/nextflow-io/rnaseq-nf/data/ggal/ggal_1_48850000_49020000.Ggal71.500bpflank.fa
 reads        : /home/boopathi/.nextflow/assets/nextflow-io/rnaseq-nf/data/ggal/ggal_gut_{1,2}.fq
 outdir       : results

WARN: [K8s] unable to fetch auth token
[-        ] process > RNASEQ:INDEX  [  0%] 0 of 1
[-        ] process > RNASEQ:INDEX                       [  0%] 0 of 1
[d8/639ec4] process > RNASEQ:FASTQC (FASTQC on ggal_gut) [100%] 1 of 1, failed: 1
[-        ] process > RNASEQ:QUANT                       -
[-        ] process > MULTIQC                            -
ERROR ~ Error executing process > 'RNASEQ:FASTQC (FASTQC on ggal_gut)'

Caused by:
  Unknown pod options: [toleration:example-key=namespace:NoSchedule]
```

Tried below options and it doesn't helped:
pod =  [[toleration: 'key:"example-key",operator:"Exists",effect:"NoSchedule"']]
pod =  [[toleration: 'key:example-key,operator:Exists,effect:NoSchedule']]
pod = [ [toleration: 'example-key=namespace:NoSchedule']]

Reference Article: https://www.nextflow.io/docs/edge/process.html#process-clusteroptions
![image](https://github.com/nextflow-io/nextflow/assets/69869823/2a256bf6-db67-4910-9117-c4361fa8fad1)

Kindly guide me on this.

**Comment by pditommaso** [comment]
Kindly format properly this comment

**Reply by Booabi** [reply]
@pditommaso - Formatted the comment. Thanks

**Reply by pditommaso** [reply]
Thanks for improving the comment formatting. @bentsherman may have more visibility on this problem

**Reply by bentsherman** [reply]
The toleration value should be a map:

```groovy
pod = [ [toleration: ['example-key': 'namespace:NoSchedule'] ]
```

Thanks for pointing this out. I think the documentation is not clear enough, so I will review these pod options and try to add some clarifications.

**Reply by bentsherman** [reply, marked as answer]
Here is a more comprehensive example with multiple toerlations:

```groovy
pod = [
    [toleration: [key: 'key1', operator: 'Equal', value: 'value1', effect: 'NoSchedule']],
    [toleration: [key: 'key1', operator: 'Exists', effect: 'NoSchedule']],
]
```

**Reply by Booabi** [reply]
Thanks for the solution @bentsherman and it works as expected. @pditommaso - Thanks for everything (Nextflow)!
